src/aspire/source/relion.py

```diff
@@ -72,12 +76,19 @@ def __init__(
 
         # Get the 'mode' (data type) - TODO: There's probably a more direct way to do this.
         mode = int(mrc.header.mode)
-        dtypes = {0: "int8", 1: "int16", 2: "float32", 6: "uint16"}
+        mrc_dtypes = {0: "int8", 1: "int16", 2: "float32", 6: "uint16"}
         assert (
-            mode in dtypes
-        ), f"Only modes={list(dtypes.keys())} in MRC files are supported for now."
+            mode in mrc_dtypes
+        ), f"Only modes={list(mrc_dtypes.keys())} in MRC files are supported for now."
 
-        dtype = dtypes[mode]
+        mrc_dtype = mrc_dtypes[mode]
+        # Potentially over ride the inferred data type.
+        if dtype is not None and dtype != mrc_dtype:
```

Comment:
Are we expecting that `dtype` is a string? In other words, do we want this warning if `dtype=np.float32` and `mrc_type='float32'`?

Reply:
I don't, and I wish most of our code didn't, but you are correct in suggesting a lot of our code allows.

I'll wrap in a np.dtype() call or something, 😇 , thanks!